[k8s Integratiion] Adding container.id as filter of Cluster Overview and removing median filter occurencies

gizas opened the pull request:
- Bug

## Proposed commit message

- WHAT: Updated the Top Memory and Top CPU visualisations of `Cluster Overview Dashboard` in order to use the **container.id** filter
- WHY:  The kubernetes.container.name used in previous visualisations was not unique for all kinds of k8s resources (eg. the kubernetes.container.name is the same for all pods of same daemonset and this was creating false calculations)

Additionally we removed all median functions from or visualisations and replaced them either with Average or Last value
The updated dashboards where the median has been removed are:
- Cronjobs
- Deployments
- Volume 
- Daemonsets


## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).


## How to test this PR locally

0. Prequisite to have a local k8s cluster. Use kind and run `kind create cluster`
1. Install kube-state-metrics in your k8s cluster
2. Clone this branch
3. Change to directory  integrations/packages/kubernetes
4. Run `elastic-package stack up -d -v --version=8.15.0` in order to spin the ES locally
5. Run `docker network connect elastic-package-stack_default kind-control-plane` to connect the networks of the ES and your k8s node
6. Open the local kibana page and install a k8s integration and create a new Elastic Agent policy
7. Apply the new manfest to your k8s cluster and once the agents are up and running navigate to your dahboards
8. Validate the changes in the visualisations

## Related issues

- Closes #10887

## Screenshots

From Cluster Overview:

![Screenshot 2024-08-27 at 12 31 56 PM](https://github.com/user-attachments/assets/cacb0461-8701-4a75-ab7c-3a7250f4e8aa)


From Cronjobs:

![Screenshot 2024-08-27 at 12 32 23 PM](https://github.com/user-attachments/assets/4de8ccf2-d060-4ce4-bbeb-957cd9b0d325)

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/15062)
* Commit: 33b209c831899f7b9580e5a65e5954545daf284d

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/15062#0191933e-2693-45e7-b3f7-ac9a827e3f74)

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/15062","number":15062,"commit":"33b209c831899f7b9580e5a65e5954545daf284d"}],"number":15062}
buildkite-pr-comment-->

elasticmachine commented:
### :rocket: Benchmarks report


#### Package `kubernetes` :&#43;1:(0) :green_heart:(0) :broken_heart:(1)
<details>
<summary>Expand to view</summary>

Data stream | Previous EPS | New EPS | Diff (%) | Result
----------- | ------------ | ------- | -------- | ------
`container_logs` | 142857.14 | 111111.11 | -31746.03 (-22.22%) | :broken_heart:
</details>


To see the full report comment with `/test benchmark fullreport`
<!--COMMENT_GENERATED_WITH_ID_benchmark-report-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/15064)
* Commit: 47d0e42c8f1e2f930d1d3b55914f5cfc80c9b5f2

### History
* :broken_heart: [Build #15062](https://buildkite.com/elastic/integrations/builds/15062) failed 33b209c831899f7b9580e5a65e5954545daf284d

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/15064","number":15064,"commit":"47d0e42c8f1e2f930d1d3b55914f5cfc80c9b5f2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/15062","number":15062,"commit":"33b209c831899f7b9580e5a65e5954545daf284d"}],"number":15064}
buildkite-pr-comment-->

constanca-m commented:
> Additionally we removed all median functions from or visualisations and replaced them either with Average or Last value

Why is this @gizas?

gizas commented:
> > Additionally we removed all median functions from or visualisations and replaced them either with Average or Last value
> 
> Why is this @gizas?

In all of the instances where median was used was a mistake, either needed to be last value or average and we aligned it with the other fields of the visualisations. Seems that because the first choice of filter is median (when someone creates a visualisation) this was left by mistake.

Additionally, after a relevant slack thread we were pointed out that average provided better results for a customer of ours so we made the decision to prefer average.

constanca-m reviewed (approved):
Alright. I always thought that median was the best one to use since it does not get affected by outliers as much as average, but I trust your judgement. Could you just link the slack thread? For future reference, it might help.

elasticmachine commented:
Package kubernetes - 1.66.4 containing this change is available at https://epr.elastic.co/search?package=kubernetes
